src/indexing.jl

```diff
@@ -145,3 +145,13 @@ function index(indexer::Indexer)
     # check if all relevant files are saved
     _check_all_files_are_saved(indexer.config.index_path)
 end
+
+function Base.show(io::IO, indexer::Indexer)
```

Comment:
For `show`ing an `Indexer`, I think you can discard configuration things like the document maxlen, compression bits, and the dimension (that should be accessible via the config). Everything else looks great; instead of calling it the `"Model"` can we call it a `"Checkpoint"` (like you did with the `ColBERTConfig`)? Also, a reference to the document collection will be great (for instance if the collection is stored in a file, use the file's path)?

Also, maybe it's a preference thing, but can we do a consistent format across all types? For instance in the `ColBERTConfig`, fields don't start with a capital letter, but for the `Indexer` they do?